source/m2mconnectionhandlerpimpl.cpp

```diff
@@ -622,6 +621,8 @@ bool M2MConnectionHandlerPimpl::is_handshake_ongoing()
 
 void M2MConnectionHandlerPimpl::receive_handler()
 {
+    unsigned char recv_buffer[BUFFER_LENGTH];
+    memset(&recv_buffer, 0, BUFFER_LENGTH);
```

Comment:
This memset() should be removed, as it hides possible uses of uninitialized memory from valgrind. I know it was there in the original version too, but it would be good to get rid of it.